scrypto-unit/src/test_runner.rs

```diff
@@ -2166,6 +2174,44 @@ impl<E: NativeVmExtension, D: TestDatabase> TestRunner<E, D> {
     }
 }
 
+pub struct DescendantVaults<'d, D> {
+    database: &'d D,
+}
+
+impl<'d, D: SubstateDatabase> DescendantVaults<'d, D> {
+    pub fn new(database: &'d D) -> Self {
+        Self { database }
+    }
+
+    pub fn get_all(&self, node_id: &NodeId) -> IndexMap<ResourceAddress, Vec<NodeId>> {
+        let mut vault_finder = VaultFinder::new();
+        let mut traverser = StateTreeTraverser::new(self.database, &mut vault_finder, 100);
+        traverser.traverse_all_descendents(None, *node_id);
```

Comment:
It seems that `traverse_all_descendents` visit the "root" node as well. 

Maybe we should give it a better name? Also impacts `DescendantVaults`.

Reply:
I'm terrible at naming. Can only think of `traverse_node_tree` and `VaultsInNodeTree`.

Reply:
`traverse_subtree()`?